docs/api-reference/dbos-compiler.md

```diff
@@ -0,0 +1,80 @@
+---
+sidebar_position: 1
+title: DBOS Compiler
+description: DBOS Compiler reference
+---
+
+The DBOS Compiler helps you deploy your DBOS applications with Stored Procedures to PostgreSQL (including your DBOS Cloud database).
+
+## Installation
+
+To install the latest Cloud CLI version for your application, run the following command in your package root:
+
+```
+npm install --save-dev @dbos-inc/dbos-compiler
+```
+
+## Stored Procedure Versioning
+
+Since [@StoredProcedure](../tutorials/stored-proc-tutorial.md) functions must be tied to a specific application version, both DBOS Transact and the DBOS Compiler are version aware. 
+By default, the application version is specified via the `DBOS__APPVERSION` environment variable, but can also be controlled via command line parameters.
+When a application version is specified, the DBOS Compiler will automatically prefix generated stored procedures with `v` and the application version.
+Likewise, DBOS Transact will automatically invoked the version prefixed version of the deployed stored procedures when the application version is specified.
+For example, if the application version was `1720214256655`, all of the stored procedures generated by would be prefixed with `v1720214256655`.
+
+:::warning
+DBOS Cloud cleans up the database deployed stored procedures for older application versions via a background task.
+Do not attempt to cleanup DBOS cloud deployed stored procedure versions yourself!
+:::
+
+## Commands
+
+---
+
+### `npx dbosc deploy`
+
+**Description:**
+This command deploys the stored procedure functions from a DBOS application to the PostgreSQL database specified in the [configuration file](./configuration).
+You must deploy your stored procedures to the database before running your DBOS application.
```

Comment:
We could specify here that `deploy` will fail if the procedures are already deployed.